upgrades/steps124_test.go

```diff
@@ -156,6 +156,31 @@ func (s *steps124Suite) TestMoveSyslogConfig(c *gc.C) {
 
 }
 
+func (s *steps124Suite) TestMoveSyslogConfigCantDeleteOld(c *gc.C) {
+	logdir := c.MkDir()
+	datadir := c.MkDir()
+	data := []byte("data!")
+	file := filepath.Join(logdir, "logrotate.conf")
+
+	// ensure that we don't error out if we can't remove the old file.
+	// error out if one of the files exists in datadir but not logdir.
+	*upgrades.OsRemove = func(string) error { return os.ErrPermission }
```

Comment:
Yeah this is a problem as it leaves the patched version in place for future execution to hit.